R/SupervisedTask.R

```diff
@@ -1,13 +1,13 @@
-makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data) {
-  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data)
+makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data, spatial) {
+  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data, spatial = spatial)
 
   if (check.data) {
     # costsens does not have a target col...
     # assertCharacter(target, any.missing = FALSE, min.len = 1L)
     w = which.first(target %nin% colnames(data))
     if (length(w) > 0L)
       stopf("Column names of data doesn't contain target var: %s", target[w])
-    checkTaskData(task$env$data, cols = setdiff(colnames(data), target))
+    checkTaskData(task$env$data, cols = setdiff(colnames(task$env$data), target))
```

Comment:
You mean e.g. benchmarking? works 👍 

Maybe there is still some confusion going on. I remove `x` and `y` from the task that arrives at the train/predict step - the original task is not touched and on every iteration it starts with the default. Hence, there should be no worries that the user will work with a modified task in other cases.

Also, if the user insists on having `x` and `y` as predictors in the modeling part, the user can just pass the coordinates as inputs which are named different than `x`/`y`, e.g. `x1` and `y1`. These variables will not be removed before the train/predict step. 
I will document all this once this PR is merged. 

```r
## Two learners to be compared
lrns = list(makeLearner("classif.lda"), makeLearner("classif.rpart"))

## Choose the resampling strategy
rdesc = makeResampleDesc("SpCV")

## Conduct the benchmark experiment
bmr = benchmark(lrns, bc.task.spatial, rdesc)

Task: basque, Learner: classif.lda
Resampling: spatial cross-validation
Measures:             mmce      
[Resample] iter 1:    0.0462963 
[Resample] iter 2:    0.2857143 
[Resample] iter 3:    0.1219512 
[Resample] iter 4:    0.2076923 
[Resample] iter 5:    0.0987654 
[Resample] iter 6:    0.1025641 
[Resample] iter 7:    0.0427350 
[Resample] iter 8:    0.3176471 
[Resample] iter 9:    0.2500000 
[Resample] iter 10:   0.2601626 


Aggregated Result: mmce.test.mean=0.1733528


Task: basque, Learner: classif.rpart
Resampling: spatial cross-validation
Measures:             mmce      
[Resample] iter 1:    0.1018519 
[Resample] iter 2:    0.2321429 
[Resample] iter 3:    0.0975610 
[Resample] iter 4:    0.1769231 
[Resample] iter 5:    0.2222222 
[Resample] iter 6:    0.0769231 
[Resample] iter 7:    0.1282051 
[Resample] iter 8:    0.3117647 
[Resample] iter 9:    0.2250000 
[Resample] iter 10:   0.2032520 
```